main.tf

```diff
@@ -1,15 +1,32 @@
 locals {
-  connect_vcs_repo = var.repository_identifier != null ? { create = true } : {}
+  connect_vcs_repo     = var.repository_identifier != null ? { create = true } : {}
+  tfc_agent_accountid  = var.tfc_agent_role_configuration["AccountId"]
+  tfc_agent_externalid = var.tfc_agent_role_configuration["ExternalId"]
 }
 
+// Add a flag to specify auth method, switch between user and role for TFC
+
 module "workspace_account" {
+  count       = var.workspace_authentication_method == "user" ? 1 : 0
   source      = "github.com/schubergphilis/terraform-aws-mcaf-user?ref=v0.1.13"
   name        = var.username
   policy      = var.policy
   policy_arns = var.policy_arns
   tags        = var.tags
 }
 
+module "workspace_account_role" {
+  count  = var.workspace_authentication_method == "role" ? 1 : 0
+  source = "github.com/schubergphilis/terraform-aws-mcaf-role?ref=v0.3.2"
+  name   = "tfc_agent"
+  assume_policy = templatefile("${path.module}/templates/assume_policy.tpl", {
+    aws_account_id = local.tfc_agent_accountid,
+    external_id    = local.tfc_agent_externalid
+  })
+  policy_arns = ["arn:aws:iam::aws:policy/AdministratorAccess"]
+  tags        = var.tags
+}
```

Comment:
Blocks should be padded and put at the end of the resource, after tags:

```suggestion
module "workspace_iam_role" {
  count       = var.workspace_authentication_method == "iam_role" ? 1 : 0
  source      = "github.com/schubergphilis/terraform-aws-mcaf-role?ref=v0.3.2"
  name        = "tfc_agent" // this should also be configurable like in the user module; it would cause problems if multiple workspaces were configured to use the same account.
  policy_arns = ["arn:aws:iam::aws:policy/AdministratorAccess"]
  tags        = var.tags
  
  assume_policy = templatefile("${path.module}/templates/assume_policy.tpl", {
    external_id    = local.tfc_agent_externalid,  // this would get updated to use output of random_string
    tfc_agent_role_arn = var.tfc_agent_role_arn,
  })
}
```

Also replaced `aws_account_id` with `tfc_agent_role_arn` in the template params as discussed.

Rather than hardcoding `["arn:aws:iam::aws:policy/AdministratorAccess"]` the module should allow us to provide an inline policy or attach an existing policy like workspace_account does (the module supports [policy_arns](https://github.com/schubergphilis/terraform-aws-mcaf-role/blob/master/variables.tf#L42-L46) and [role_policy](https://github.com/schubergphilis/terraform-aws-mcaf-role/blob/master/variables.tf#L54-L58))